strapi/strapi
# Regenerate an API token

What does it do?
As a user who has access to the settings of a token, I want to have the possibility to regenerate a token, so that I can reset its value if needed.

👨🏼‍🎨 Design
[https://www.figma.com/file/lADY9wlnGvOX1blL19UYcG/%5BWIP%5D-DX---API-Tokens?node-id=1485%3A8952 - Connect to preview](https://www.figma.com/file/lADY9wlnGvOX1blL19UYcG/%5BWIP%5D-DX---API-Tokens?node-id=1485%3A8952) 

[https://www.figma.com/file/lADY9wlnGvOX1blL19UYcG/%5BWIP%5D-DX---API-Tokens?node-id=1485%3A11608 - Connect to preview](https://www.figma.com/file/lADY9wlnGvOX1blL19UYcG/%5BWIP%5D-DX---API-Tokens?node-id=1485%3A11608)

How to test it?
Should satisfy the following criterias:

- The “Regenerate” button is visible only when I create a token (once I save and I successfully create the token) and when I edit the settings of a token
 - When I click on the “Regenerate” button
   - then a modal appear asking for confirmation
   - when I click regenerate
     - then the token is regenerated and displayed following the design
   - when I click on cancel
     - then I come back to the token settings
